packages/millicast-sdk/src/utils/Codecs.js

```diff
@@ -27,3 +29,575 @@ export const AudioCodec = {
   OPUS: 'opus',
   MULTIOPUS: 'multiopus'
 }
+
+const NALUType = {
+  SLICE_NON_IDR: 1,
+  SLICE_PARTITION_A: 2,
+  SLICE_IDR: 5,
+  SEI_H264: 6,
+  SEI_H265_PREFIX: 39,
+  SEI_H265_SUFFIX: 40,
+  SPS_H264: 7,
+  SPS_H265: 33,
+  PPS_H264: 8,
+  PPS_H265: 34
+}
+
+const SEI_Payload_Type = {
+  PIC_TIMING: 1,
+  USER_DATA_UNREGISTERED: 5
+}
+
+class SPSState {
+  constructor (codec = 'h264') {
+    this.sps = new Map()
+    this.pps = new Map()
+    this.activeSPS = null
+    this.codec = codec
+  }
+
+  collectPPS (rbsp) {
+    if (this.codec === 'h264') {
+      this.collectH264PPS(rbsp)
+    } else {
+      this.collectH265PPS(rbsp)
+    }
+  }
+
+  collectSPS (rbsp) {
+    if (this.codec === 'h264') {
+      this.collectH264SPS(rbsp)
+    } else {
+      this.collectH265SPS(rbsp)
+    }
+  }
+
+  collectH264SPS (rbsp) {
+    const reader = new BitStreamReader(rbsp)
+    const profile_idc = reader.readBits(8)
+    reader.skip(8) // skip 8bits constraint set flag and reserved_zero_2bits
+    reader.skip(8) // level_idc
+    const seq_parameter_set_id = reader.readExpGolombUnsigned()
+    if (seq_parameter_set_id > 31 || seq_parameter_set_id < 0) {
+      throw new Error('Invalid seq_parameter_set_id')
+    }
+    if ([100, 110, 122, 244, 44, 83, 86, 118, 128, 138, 139, 134, 135].includes(profile_idc)) {
```

Comment:
nit : can we extract this array into a const for readability? 

`const supported_profiles` or whatever

Reply:
I will also refactor variable names to use standard camel case